packages/grafana-data/src/transformations/fieldReducer.ts

```diff
@@ -419,6 +435,36 @@ function calculateLastNotNull(field: Field, ignoreNulls: boolean, nullAsZero: bo
   return { lastNotNull: null };
 }
 
+function calculateVariance(field: Field, ignoreNulls: boolean, nullAsZero: boolean): FieldCalcs {
+  let squareSum = 0;
+  let runningMean = 0;
+  let runningNonNullCount = 0;
+  let variance = null;
+  const isNumberField = field.type === FieldType.number || FieldType.time;
+  const data = field.values;
+  for (let i = 0; i < data.length; i++) {
+    let currentValue = data.get(i);
+    if (currentValue != null) {
+      if (isNumberField) {
+        runningNonNullCount++;
+        let _oldMean = runningMean;
+        runningMean += (currentValue - _oldMean) / runningNonNullCount;
+        squareSum += (currentValue - _oldMean) * (currentValue - runningMean);
+      }
+    }
+  }
+  if (runningNonNullCount > 0) {
+    variance = squareSum / runningNonNullCount;
+  }
+  return { variance: variance };
```

Comment:
```suggestion
  return { variance: variance, stdDev: Math.sqrt(variance) };
```

and then remove the second function.  This will let both values be calculated at the same time.

Reply:
Hi @ryantxu. I have made change as per your suggestion.